src/components/POLine/POLineInvoices/POLineInvoicesContainer.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { stripesConnect } from '@folio/stripes/core';
+import { Accordion } from '@folio/stripes/components';
+
+import {
+  INVOICE_LINES,
+  INVOICES,
+  RECEIVING_HISTORY,
+} from '../../Utils/resources';
+
+import POLineInvoices from './POLineInvoices';
+
+class POLineInvoicesContainer extends Component {
+  static manifest = Object.freeze({
+    pieces: RECEIVING_HISTORY,
+    invoiceLines: INVOICE_LINES,
+    invoices: INVOICES,
+  });
+
+  static propTypes = {
+    label: PropTypes.object.isRequired,
+    accordionId: PropTypes.string.isRequired,
+    mutator: PropTypes.shape({
+      pieces: PropTypes.object.isRequired,
+      invoices: PropTypes.object.isRequired,
+      invoiceLines: PropTypes.object.isRequired,
+    }).isRequired,
+    lineId: PropTypes.string.isRequired,
+    vendors: PropTypes.arrayOf(PropTypes.object),
+    quantityPhysical: PropTypes.number,
+    quantityElectronic: PropTypes.number,
+  };
+
+  static defaultProps = {
+    vendors: [],
+    quantityPhysical: 0,
+    quantityElectronic: 0,
+  };
+
+  state = {
+    lineInvoices: [],
+    pieces: [],
+  }
+
+  componentDidMount() {
+    this.fetchLineInvoices();
+    this.fetchPieces();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.lineId !== prevProps.lineId) {
+      this.fetchLineInvoices();
+      this.fetchPieces();
+    }
+  }
+
+  fetchLineInvoices = () => {
+    const { mutator, lineId } = this.props;
+
+    mutator.invoiceLines.reset();
+    mutator.invoiceLines.GET({
+      params: {
+        query: `poLineId==${lineId}`,
+      },
+    }).then(list => list.map(item => item.invoiceId))
+      .then(invoicesIds => {
+        mutator.invoices.reset();
+
+        return invoicesIds.length
+          ? mutator.invoices.GET({
+            params: {
+              query: invoicesIds.map(id => `id==${id}`).join(' or '),
+            },
+          })
+          : [];
+      })
+      .then(lineInvoices => this.setState({ lineInvoices }))
```

Comment:
what is the reason to keep that in state?

Reply:
Rewrote using resources